ggshield/cmd/sca/scan/ci.py

```diff
@@ -0,0 +1,75 @@
+from pathlib import Path
+from typing import Any, Optional, Sequence
+
+import click
+
+from ggshield.cmd.common_options import directory_argument
+from ggshield.cmd.sca.scan.sca_scan_utils import (
+    create_output_handler,
+    display_sca_beta_warning,
+    sca_scan_all,
+    sca_scan_diff,
+    update_context,
+)
+from ggshield.cmd.sca.scan.scan_common_options import add_sca_scan_common_options
+from ggshield.cmd.utils import get_commits
+from ggshield.core.errors import handle_exception
+from ggshield.sca.collection.collection import (
+    SCAScanAllVulnerabilityCollection,
+    SCAScanDiffVulnerabilityCollection,
+)
+
+
+@click.command()
+@add_sca_scan_common_options()
+@click.pass_context
+@directory_argument
+@click.option(
+    "--all",
+    help="Scan all files.",
+    is_flag=True,
+)
+@display_sca_beta_warning
+def ci_cmd(
+    ctx: click.Context,
+    exit_zero: bool,
+    minimum_severity: str,
+    ignore_paths: Sequence[str],
+    directory: Optional[Path],
+    all,
+    **kwargs: Any,
+) -> int:
+    """
+    scan in a CI environment.
+    """
+    if directory is None:
+        directory = Path().resolve()
+
+    # Adds client and required parameters to the context
+    update_context(ctx, exit_zero, minimum_severity, ignore_paths)
+
+    config = ctx.obj["config"]
+    try:
+
+        output_handler = create_output_handler(ctx)
+        if all:
+            result = sca_scan_all(ctx=ctx, directory=directory)
+            scan = SCAScanAllVulnerabilityCollection(id=str(directory), result=result)
+            return output_handler.process_scan_all_result(scan)
+
+        else:
+            commit_list, _ = get_commits(config)
+
+            if config.verbose:
+                click.echo(f"Commits to scan: {len(commit_list)}", err=True)
+            result = sca_scan_diff(
+                ctx=ctx,
+                directory=directory,
+                ref=f"HEAD~{len(commit_list)}",
```

Comment:
In an MR for example, this will ignore the current state of the branch on which we want to merge. 
@GG-HH, is this the expected behavior ? I know there has been some discussions about this, and at some point we wanted to use the targeted branch state as a ref in a MR.

Reply:
Let's merge this behavior, we might discuss this when Loic is back.